src/core/api.c

```diff
@@ -546,7 +546,7 @@ mender_api_check_for_deployment(mender_api_deployment_data_t *deployment) {
         ret = MENDER_NOT_FOUND;
     } else {
         mender_api_print_response_error(response, status);
-        ret = MENDER_FAIL;
+        ret = MENDER_NET_ERROR;
```

Comment:
Same here and below -- if we get a response and have a status, it's not a network error.

Reply:
(maybe this was discussed outside?)

Even when getting an status, we will retry with the retry interval. This is specially important for 401 Unauthorized, as you don't want to wait the full poll interval after you sent your first auth request. Other status like rate limiting or server errors should also follow this logic. This was a painful lesson learned from Mender Client 3 to Mender Client4/5, and there we ended up by on all error responses.